test/system/mentor/dashboard_test.rb

```diff
@@ -0,0 +1,25 @@
+require 'application_system_test_case'
+
+class DashboardTest < ApplicationSystemTestCase
+  test "displays the number of days ago a solution was updated" do
+    track = create(:track)
+    mentor = create(:user_mentor,
+                    accepted_terms_at: Date.new(2016, 12, 25),
+                    accepted_privacy_policy_at: Date.new(2016, 12, 25),
+                    mentored_tracks: [track])
+    exercise = create(:exercise, title: "Exercise 1", track: track)
+    solution = create(:solution, exercise: exercise,
+                mentoring_requested_at: Time.current,
+                last_updated_by_user_at: Time.now - 38.days)
+    create(:iteration, solution: solution)
+    create(:solution_mentorship,
+           user: mentor,
+           solution: solution,
+           requires_action_since: Date.yesterday)
```

Comment:
Here the key information that the reader is looking for is mixed into all the boilerplate. To make it as easy as possible for the reader of the test, I would ask myself what would be critical to know in order to read the assertion and be certain that it's testing the right thing?

I can't tell if the dates on lines 7 and 8 are important (probably not), and also I'm not sure why they're chosen as Christmas day. If they're not important I would pick something simpler, like January 1st. I might also put it in a local variable named:

```
some_date = Date.new(2016, 1, 1)
```

I also can't tell if it's important that the mentoring was requested right now (line 12), or when it requires action since (line 18). On the other hand, I'm puzzled at the fact that the solution was last updated 38 days ago, but only requires action since yesterday. I'm guessing that those two times are not important for the test, but I would try to avoid making the reader have to think too hard about it.

My understanding after having read through this, is that the only thing that _is_ important is the timestamp on line 13.
Since that timestamp is important, I would put it in a local variable at the very top.

That would make this something like:

```
solution_updated_at = Time.now - 38.days

unimportant_date = Date.new(2016, 1, 1)
unimportant_time = Time.current
track = create(:track)
mentor = create(:user_mentor, ...)
exercise = create(:exercise, ...)
solution = create(:solution, ...)
create(:iteration, ...)
create(:solution_mentorship, ...)

sign_in!(mentor)
visit mentor_dashboard_path

assert page.has_content?('38 days ago')
```